### PR TITLE
Prevents updating qmore configuration on every job reservation.

### PR DESCRIPTION
**Pull request by james-lawrence**
Prevents updating configuration every call to reserve a job by offloading the update into a background thread.
Still testing.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -14,12 +14,19 @@ Alternatively, if you have some other way of launching workers (e.g. qless-pool)
 
     Qless::Pool.pool_factory.reserver_class = Qmore::JobReserver
     Qmore.client == Qless::Pool.pool_factory.client
+    # Redis persistance defaults to using the Qmore.client.
+    Qmore.persistance = Qless::Persistance::Redis.new(Qmore.client.redis)
```
Comment (wr0ngway) [inline README.md]: persistence not ance

File: README.md [hunk 2]
```diff
@@ -14,12 +14,19 @@ Alternatively, if you have some other way of launching workers (e.g. qless-pool)
 
     Qless::Pool.pool_factory.reserver_class = Qmore::JobReserver
     Qmore.client == Qless::Pool.pool_factory.client
+    # Redis persistance defaults to using the Qmore.client.
```
Comment (wr0ngway) [inline README.md]: probably should be more explicit and mention that two redis connections are needed (can be the same) - persistence is for configuration, and client is still needed when fetching the queues

File: README.md [hunk 3]
```diff
@@ -14,12 +14,19 @@ Alternatively, if you have some other way of launching workers (e.g. qless-pool)
...
+    Qmore.persistance = Qless::Persistance::Redis.new(Qmore.client.redis)
+    # Configure the monitor thread with the persistance type, and the interval at which to update
+    # Monitor defaults to using Qmore.persistance and 2 minutes
+    Qmore.monitor = Qless::persistance::Monitor.new(Qmore.persistance, 120)
```
Comment (wr0ngway) [inline README.md]: Would it make more sense to keep the monitor as part of the persistence object?  Seems like a bit of a leaky abstraction - got 3 objects inter-relating

File: README.md [hunk 4]
```diff
@@ -14,12 +14,19 @@ Alternatively, if you have some other way of launching workers (e.g. qless-pool)
...
+    # Start up monitor thread
+    Qmore.monitor.start
```
Comment (wr0ngway) [inline README.md:28]: The average case is probably a single redis server for both client and persistence, with some sane default for monitoring/updating config - can we make that work without the additional setup - i.e. keep things working the way there were before this change for existing users, but have the ability to configure persistence/monitoring

Comment (james-lawrence) [inline README.md:28]: yup. It'll just be a wrapper Persistance which delegates down.

File: README.md (hunk 1 above)
Comment (james-lawrence) [inline README.md]: well that is embarrassing

File: README.md (hunk 3 above)
Comment (james-lawrence) [inline README.md]: Not really. having the separate class for monitoring allows more flexibility.
For example if I write a file based persistence object how I have to include how to monitor it as well?
Same goes if I want a different monitoring strategy for example publish/subscribe. Now I need to update every persistence class?

They are two very different responsibilities.

Comment (wr0ngway) [inline README.md]: I wasn't arguing against separate objects, just wondering if Qmore needed the ref to the monitor, i.e.
Qmore.monitor moves to persistence.monitor.

Comment (james-lawrence) [inline README.md]: ah. hmm let me think on that.

**Pull request merged**